tests/providers/aws/services/sns/sns_topics_not_publicly_accessible/sns_topics_not_publicly_accessible_test.py

```diff
@@ -226,3 +239,37 @@ def test_topic_public(self):
             assert result[0].resource_arn == topic_arn
             assert result[0].region == AWS_REGION_EU_WEST_1
             assert result[0].resource_tags == []
+
+    def test_topic_public_with_principal_org_id(self):
+        sns_client = mock.MagicMock
+        sns_client.audited_account = AWS_ACCOUNT_NUMBER
+        sns_client.topics = []
+        sns_client.topics.append(
+            Topic(
+                arn=topic_arn,
+                name=topic_name,
+                policy=test_policy_restricted_principal_org_id,
+                region=AWS_REGION_EU_WEST_1,
+                org_id=org_id,
+            )
+        )
+        with mock.patch(
+            "prowler.providers.aws.services.sns.sns_service.SNS",
+            sns_client,
+        ):
+            from prowler.providers.aws.services.sns.sns_topics_not_publicly_accessible.sns_topics_not_publicly_accessible import (
+                sns_topics_not_publicly_accessible,
+            )
+
+            check = sns_topics_not_publicly_accessible()
+            result = check.execute()
+            assert len(result) == 1
+            assert result[0].status == "PASS"
+            assert (
+                result[0].status_extended
+                == f"SNS topic {topic_name} is not public because its policy only allows access from the same account."
```

Comment:
We will have to add another status, if they have an organization in the conditions it has cross-account access for all the accounts inside the org.

Reply:
Good catch!

Reply:
I'd replace "from the same account" with "from a specific org or specific accounts"

Reply:
Agree with @mlanders-spectra.